controllers/function_controller.go

```diff
@@ -29,15 +30,19 @@ import (
 	"k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/runtime"
 	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
 // FunctionReconciler reconciles a Function object
 type FunctionReconciler struct {
 	client.Client
-	Log    logr.Logger
-	Scheme *runtime.Scheme
+	Log                           logr.Logger
+	Scheme                        *runtime.Scheme
+	functionGenerations           *sync.Map
+	isFunctionGenerationIncreased bool
```

Comment:
looks like the `functionGenerations` stores all function generations separately using a map, while `isFunctionGenerationIncreased` is shared for all functions. will this cause any conflicts?

and when the FM controller pod is recreated, these data will be lost, does this have any effects for the reconciliation?

Reply:
You're right, I need to recheck the `isFunctionGenerationIncreased` issue.

If the controller restarts, a routine reconciliation is performed and the Generations state of all resources is restored.